plot-lib/plot/private/no-gui/plot2d.rkt

```diff
@@ -68,7 +70,16 @@
                     [plot-legend-anchor  legend-anchor])
        (define area (make-object 2d-plot-area%
                       bounds-rect x-ticks x-far-ticks y-ticks y-far-ticks legend dc x y width height))
-       (plot-area area renderer-list))]))
+       (plot-area area renderer-list)
+
+       (define bounds (vector (vector (assert (ivl-min (vector-ref bounds-rect 0)) real?)
+                                      (assert (ivl-max (vector-ref bounds-rect 0)) real?))
+                              (vector (assert (ivl-min (vector-ref bounds-rect 1)) real?)
+                                      (assert (ivl-max (vector-ref bounds-rect 1)) real?))))
+       (new (class object%
```

Comment:
While creating a new class type and immediately instantiating an object is supported by Racket, I am not sure if this is the most efficient way to do it in terms of memory allocation and runtime performance -- this object will be discarded in most current uses of the plot package).

@samth do you know whether it would be more efficient to create a separate `plot-metrics%` class and just instantiate this here?

Reply:
@mflatt will know better here, but it will at most save a bit of allocation here.

Reply:
We only have one use case for these changes (https://github.com/racket/plot/issues/83#issuecomment-790017560) and this would involve calling `plot/dc` several times a second (30FPS perhaps).  I think this code should avoid any unnecessary memory allocations and defer any deferrable computations until they are needed (e.g. calculating the bounds).